### PR TITLE
Balancer `dex.trades_beta` exposed views

### PR DESCRIPTION
**Pull request by tomfutago**
follow up PR to expose views built on top of `dex.trades_beta` + existing logic for backward compatibility

### CODE REVIEW

**Comment by jeff-dude** [comment]
gm gm @viniabussafi @thetroyharris 

we're rebuilding the `dex` sector lineage, and part of that was simplifying initial read for balancer data that eventually loads into `dex.trades` (not just balancer, all projects). balancer is an outlier in terms of additional columns being pulled into the project-specific trades spells in the old `dex.trades` lineage. to make this all work in the new design, our plan is:
- keep initial read of balancer data that feeds `dex.trades_beta` simple and match all other projects
- once `dex.trades_beta` is built, add views on top for project-specific spells that want extra logic

tom has been generous enough to help us migrate and has put out initial build of the views here in this PR. when you guys get the time (no rush), plz do review this here and give us approval if they work for you 🙏  eventually, we'll be removing `_beta` from both `dex.trades` and these views, so they will replace the old models

**Comment by viniabussafi** [comment]
Hey @jeff-dude and @tomfutago!
Thanks for your work on this! This transition you're making in the dex.trades model is looking extremely clean and I'm looking forward to seeing the next steps.

> keep initial read of balancer data that feeds dex.trades_beta simple and match all other projects
Just to make sure we're on the same page, would this mean eventually removing columns like `pool_id` and `swap_fee` from the `balancer.trades` model?

Code-wise, I like what you built and have absolutely no objection to it as-is.

**Comment by viniabussafi** [comment]
Tagging @mendesfabio here for review as well

**Comment by tomfutago** [comment]
> Hey @jeff-dude and @tomfutago! Thanks for your work on this! This transition you're making in the dex.trades model is looking extremely clean and I'm looking forward to seeing the next steps.
> 
> > keep initial read of balancer data that feeds dex.trades_beta simple and match all other projects
> 
> Just to make sure we're on the same page, would this mean eventually removing columns like `pool_id` and `swap_fee` from the `balancer.trades` model?
> 
> Code-wise, I like what you built and have absolutely no objection to it as-is.

Hey @viniabussafi, thanks for reviewing and kind words! 

Re `pool_id` and `swap_fee` columns - the plan is to keep them exactly where they are (`balancer.trades` and related models under `balancer` project) and this PR is a move in that direction. Hope that clarifies.

**Comment by viniabussafi** [comment]
> Re pool_id and swap_fee columns - the plan is to keep them exactly where they are (balancer.trades and related models under balancer project) and this PR is a move in that direction. Hope that clarifies.

Sounds great! AFAIC, you can go on with this PR.

**Review by viniabussafi — APPROVED** [review]

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @tomfutago 

[Balancer_v2_*.base_trades checking](https://dune.com/queries/3314418)

File: models/balancer/arbitrum/balancer_v2_arbitrum_trades_beta.sql [hunk 1]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_arbitrum',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/arbitrum/balancer_v2_arbitrum_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["arbitrum"]\',
```

File: models/balancer/avalanche_c/balancer_v2_avalanche_c_trades_beta.sql [hunk 2]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_avalanche_c',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/avalanche_c/balancer_v2_avalanche_c_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["avalanche_c"]\',
```

File: models/balancer/base/balancer_v2_base_trades_beta.sql [hunk 3]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_base',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/base/balancer_v2_base_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["base"]\',
```

File: models/balancer/ethereum/balancer_v2_ethereum_trades_beta.sql [hunk 4]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_ethereum',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/ethereum/balancer_v2_ethereum_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["ethereum"]\',
```

File: models/balancer/gnosis/balancer_v2_gnosis_trades_beta.sql [hunk 5]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_gnosis',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/gnosis/balancer_v2_gnosis_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["gnosis"]\',
```

File: models/balancer/optimism/balancer_v2_optimism_trades_beta.sql [hunk 6]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_optimism',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/optimism/balancer_v2_optimism_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["optimism"]\',
```

File: models/balancer/polygon/balancer_v2_polygon_trades_beta.sql [hunk 7]
```diff
@@ -0,0 +1,138 @@
+{{
+    config(
+        schema = 'balancer_v2_polygon',
+        alias = 'trades_beta',
+        materialized = 'view',
+        post_hook = '{{ expose_spells_hide_trino(\'["celo"]\',
```
Comment (jeff-dude) [inline models/balancer/polygon/balancer_v2_polygon_trades_beta.sql:6]:
```suggestion
        post_hook = '{{ expose_spells(\'["polygon"]\',
```

**Review by jeff-dude — CHANGES_REQUESTED** [review]
in addition to below config fixes, i have another question. thinking out loud here.

can we skip the beta phase on these views? if we are confident `dex.trades_beta` matches `dex.trades` for balancer specifically, we can just make these views the official ones? then in the future, when we deprecate `_beta` in dex, we will be updating all refs in spellbook anyway

let me know if i'm missing something as to why we can't make these official now

**Comment by tomfutago** [comment]
> in addition to below config fixes, i have another question. thinking out loud here.
> 
> can we skip the beta phase on these views? if we are confident `dex.trades_beta` matches `dex.trades` for balancer specifically, we can just make these views the official ones? then in the future, when we deprecate `_beta` in dex, we will be updating all refs in spellbook anyway
> 
> let me know if i'm missing something as to why we can't make these official now

I've updated configs.

With beta phase views - I'd vote to keep them in for now - for the following reasons:
* `dex.trades_beta` actually doesn't match `dex.trades` for balancer because of added logic on `amount_usd` column, so having beta views would make comparing to current models easier (both for me and/or Balancer team)
* hopefully not an issue at all, but would be good to compare performance of existing models to beta views - with main difference being that current models are incremental on `bpt_prices`, while beta views are just that - straight up views
* final one - it might be already resolved, but I think there are still cases where `amount_usd` is blank in `dex.trades_beta` - unless that's all back-populated now? [ref note](https://github.com/duneanalytics/spellbook/issues/4759#issuecomment-1845887403)

**Comment by jeff-dude** [comment]
> With beta phase views - I'd vote to keep them in for now - for the following reasons:
> 
> * `dex.trades_beta` actually doesn't match `dex.trades` for balancer because of added logic on `amount_usd` column, so having beta views would make comparing to current models easier (both for me and/or Balancer team)
> * hopefully not an issue at all, but would be good to compare performance of existing models to beta views - with main difference being that current models are incremental on `bpt_prices`, while beta views are just that - straight up views
> * final one - it might be already resolved, but I think there are still cases where `amount_usd` is blank in `dex.trades_beta` - unless that's all back-populated now? [ref note](https://github.com/duneanalytics/spellbook/issues/4759#issuecomment-1845887403)

this all sounds good to me. thanks for the context.

as for `amount_usd` gaps, our ingest team noticed that the prices backfill job was accidentally paused. this is the job that fills in missing minute gaps. my understanding is that `prices.usd` gaps have been recovered, we would just need a refresh on `dex.trades` and all other downstream spells that use it. i actually think we've had a refresh on `dex.trades` this week, so we may see some gaps covered.

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**